FirebaseInstallations/Source/Library/FIRInstallations.m

```diff
@@ -117,12 +119,40 @@ - (instancetype)initWithAppOptions:(FIROptions *)appOptions
   return self;
 }
 
++ (void)validateAppOptions:(FIROptions *)appOptions appName:(NSString *)appName {
+  NSMutableArray *missingFields = [NSMutableArray array];
+  if (appName.length < 1) {
+    [missingFields addObject:@"`FirebaseApp.name`"];
+  }
+  if (appOptions.APIKey.length < 1) {
+    [missingFields addObject:@"`FirebaseOptions.APIKey`"];
+  }
+  if (appOptions.googleAppID.length < 1) {
+    [missingFields addObject:@"`FirebaseOptions.googleAppID`"];
+  }
+  if (appOptions.GCMSenderID.length < 1) {
+    [missingFields addObject:@"`FirebaseOptions.GCMSenderID`"];
+  }
+
+  if (missingFields.count > 0) {
+    [NSException
+         raise:kFirebaseInstallationsErrorDomain
+        format:@"%@[%@] Could not configure Firebase Installations due to invalid FirebaseApp "
+               @"options. The following parameters are nil or empty: %@. If you use "
+               @"GoogleServices-Info.plist please download the most recent version from Firebase "
```

Comment:
nit: include the word "the" --> "from the Firebase console"

Reply:
updated